public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/PartListSection.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { cx, css } from '@emotion/css';
+import { SegmentInput, MenuItem, WithContextMenu, MenuGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { Seg } from './Seg';
+import { unwrap } from './unwrap';
+import { toSelectableValue } from './toSelectableValue';
+import { AddButton } from './AddButton';
+
+export type PartParams = Array<{
+  value: string;
+  options: (() => Promise<string[]>) | null;
+}>;
+
+type Props = {
+  parts: Array<{
+    name: string;
+    params: PartParams;
+  }>;
+  getNewPartOptions: () => Promise<SelectableValue[]>;
+  onChange: (partIndex: number, paramValues: string[]) => void;
+  onRemovePart: (index: number) => void;
+  onAddNewPart: (type: string) => void;
+};
+
+const renderRemovableNameMenuItems = (onClick: () => void) => {
+  return (
+    <MenuGroup label="" ariaLabel="">
+      <MenuItem label="remove" ariaLabel="remove" onClick={onClick} />
+    </MenuGroup>
+  );
+};
+
+const noRightMarginPaddingClass = css({
+  paddingRight: '0',
+  marginRight: '0',
+});
+
+const RemovableName = ({ name, onRemove }: { name: string; onRemove: () => void }) => {
+  return (
+    <WithContextMenu renderMenuItems={() => renderRemovableNameMenuItems(onRemove)}>
+      {({ openMenu }) => (
+        <button className={cx('gf-form-label', noRightMarginPaddingClass)} onClick={openMenu}>
+          {name}
+        </button>
+      )}
+    </WithContextMenu>
+  );
+};
+
+type PartProps = {
+  name: string;
+  params: PartParams;
+  onRemove: () => void;
+  onChange: (paramValues: string[]) => void;
+};
+
+const noLeftPaddingClass = css({
+  paddingLeft: '0',
+});
+
+const noHorizMarginPaddingClass = css({
+  paddingLeft: '0',
+  paddingRight: '0',
+  marginLeft: '0',
+  marginRight: '0',
+});
+
+const Part = ({ name, params, onChange, onRemove }: PartProps): JSX.Element => {
+  const onParamChange = (par: string, i: number) => {
+    const newParams = params.map((p) => p.value);
+    newParams[i] = par;
+    onChange(newParams);
+  };
+  return (
+    <div className="gf-form-inline">
+      <div className={cx('gf-form-label', noLeftPaddingClass)}>
```

Comment:
i think `gf-form-label` is making the options inside the segment look a bit off (too big, I think it's setting `line-height: 32px;` ):
<img width="233" alt="Screenshot 2021-05-04 at 16 46 53" src="https://user-images.githubusercontent.com/1170767/117031298-57cf4380-acf8-11eb-91c3-88ea1481e9df.png">

compared to the one in normal selects/other components:

<img width="211" alt="Screenshot 2021-05-04 at 16 47 29" src="https://user-images.githubusercontent.com/1170767/117031381-6d446d80-acf8-11eb-95f0-3a5e4b4f5c27.png">

Reply:
nice find! fixed.